lib/cea/cea_decoder.js

```diff
@@ -0,0 +1,293 @@
+          ccPacket.type === shaka.cea.CeaDecoder.NTSC_CC_FIELD_2) {
+        const parsedClosedCaption = this.decodeCea608_(ccPacket);
+        if (parsedClosedCaption) {
+          parsedClosedCaptions.push(parsedClosedCaption);
+        }
+      }
+    }
+
+    this.clearExtractedPackets_();
+    return parsedClosedCaptions;
+  }
+
+  /**
+   * Decodes a CEA-608 closed caption packet based on ANSI/CEA-608.
+   * @param {shaka.cea.Cea608DataChannel.Cea608Packet} ccPacket
+   * @return {?shaka.cea.ICaptionDecoder.ClosedCaption}
+   * @private
+   */
+  decodeCea608_(ccPacket) {
+    const fieldNum = ccPacket.type;
+    let b1 = ccPacket.ccData1;
+    let b2 = ccPacket.ccData2;
+
+    // If this packet is a control code, then it also sets the channel.
+    // For control codes, cc_data_1 has the form |P|0|0|1|C|X|X|X|.
+    // "C" is the channel bit. It indicates whether to set C2 active.
+    if (shaka.cea.Cea608DataChannel.isControlCode(b1)) {
+      const channelNum = (b1 >> 3) & 0x01; // Get channel bit.
+
+      // Change the stream based on the field, and the new channel
+      if (fieldNum === 0) {
+        this.currentField1Channel_ = channelNum;
+      } else {
+        this.currentField2Channel_ = channelNum;
+      }
+    }
+
+    // Get the correct stream for this caption packet (CC1, ..., CC4)
+    const selectedChannel = fieldNum ?
+        this.currentField2Channel_ : this.currentField1Channel_;
+    const selectedMode = `CC${(fieldNum << 1) | selectedChannel + 1}`;
+    const selectedStream = this.cea608ModeToStream_.get(selectedMode);
+
+    // If on field 1, check if there are too many bad frames
+    if (fieldNum === 0) {
+      if ((b1 === 0xff && b2 === 0xff) || (!b1 && !b2)) {
+        // Completely invalid frame.  FCC says after 45 of them, reset.
+        if (++this.badFrames_ >= 45) {
+          this.reset();
+        }
+      }
+    }
+
+    // Validate the parity, cc_data_1 and cc_data_2 should be odd parity uimsbf.
+    if (!this.isOddParity_(b1) || !this.isOddParity_(b2)) {
+      return null;
+    }
+
+    // Remove the MSB (parity bit).
+    ccPacket.ccData1 = (b1 &= 0x7f);
+    ccPacket.ccData2 = (b2 &= 0x7f);
+
+    // Check for empty captions and skip them.
+    if (!b1 && !b2) {
+      return null;
+    }
+
+    // Process the clean CC data pair.
+    let parsedClosedCaption = null;
+    if (shaka.cea.Cea608DataChannel.isControlCode(b1)) {
+      parsedClosedCaption = selectedStream.handleControlCode(ccPacket);
+    } else {
+      // Handle as a Basic North American Character.
+      selectedStream.handleBasicNorthAmericanChar(b1, b2);
+    }
+
+    return parsedClosedCaption;
+  }
+
+  /**
+   * Checks if the provided byte has odd parity.
+   * @param {!number} byte
+   * @return {!boolean} True if the byte has odd parity.
+   * @private
+   */
+  isOddParity_(byte) {
+    let parity = 0;
+    while (byte) {
+      parity ^= 1;
+      byte &= (byte - 1);
```

Comment:
I find this calculation hard to comprehend.  At least, I don't find it intuitive what "thing bitwise-and one less than itself" really does.  I ran a few examples of this in my head and based on the output, it appears that you're counting the number of "1" bits.

If that's the goal (check for odd number of 1 bits), I'd find this clearer:

```js
let parity = 0;
for (let i = 0; i < 8; ++i) {  // 8-bit byte as input
  parity ^= (byte & 1);  // toggle parity if low bit is 1
  byte >>= 1;  // shift away the low bit
}
return parity == 1;
```

Or even with `while (byte) {` as the loop instead of i from 0 to 8, if you prefer.